web/web/files--common/javascript/OZONE/dialog.ts

```diff
@@ -0,0 +1,675 @@
+    if (!el) {
+      el = document.createElement('div');
+      el.id = 'odialog-hovertips';
+      el.style.position = "absolute";
+      el.style.zIndex = "100";
+      el.style.top = "0";
+      el.style.width = "100%";
+      const body = document.getElementsByTagName('body')[0];
+      body.appendChild(el);
+      dialog.hovertip.container = el;
+    }
+  },
+
+  makeTip: function (
+    element: string | HTMLElement | HTMLCollection,
+    options?: HovertipOptions
+  ): void {
+    /**
+     * ??
+     *
+     * @param element: The element ??
+     * Can be given as a string representing an element's ID, a reference to
+     * an element itself, or an array of such references as returned by e.g.
+     * getElementsByTagName().
+     * @param options: Settings for the hovertip. May contain either context
+     * or text, but should not contain both.
+     *   context: The ID of the element to ??
+     *   text: Text of the hovertip.
+     *   style: Object containing style rules for the hovertip.
+     *   noCursorHelp: ??
+     *   delay: ??
+     *   smartWidthLimit: ??
+     *   valign: ??
+     */
+    // XXX What is 'el'? It's not the element representing the hovertip
+    // itself - right? Is it the element that will show the hovertip once
+    // hovered?
+    // XXX What is 'tipEl'? Is that the hovertip?
+    // TODO Rename those variables once we know
+
+    // Type guard to check if the 'element' is a collection
+    const isHtmlCollection = (
+      element: HTMLElement | HTMLCollection
+    ): element is HTMLCollection => 'length' in element;
+
+    let el: HovertipElement;
+    if (typeof element !== "string") {
+      if (isHtmlCollection(element)) {
+        // Recurse for each element in the array
+        Array.from(element).forEach(element => {
+          // XXX Type assertions like this are indicative of bad functions
+          // TypeScript thinks `element` is type Element
+          const htmlElement = element as HTMLElement;
+          dialog.hovertip.makeTip(htmlElement, options);
+        });
+        return;
+      } else {
+        el = element;
+      }
+    } else {
+      el = document.getElementById(element)!;
+    }
+
+    // options can be: text (text) or context element id (context)
+    dialog.hovertip.init(); // just for sure
+    if (!el) {
+      return;
+    }
+    if (el.hovertip) {
+      // XXX What is el.hovertip? Why is it needed? Where is it assigned?
+      return;
+    }
+
+    let tipEl: HTMLElement;
+    if (options && options.context) {
+      // Context contains an existing ID to use as tipEl
+      const el = document.getElementById(options.context);
+      if (el === null) {
+        // XXX This should probably throw an error
+        return;
+      }
+      tipEl = el;
+    } else if (options && options.text) {
+      // create a new div
+      tipEl = document.createElement('div');
+      tipEl.innerHTML = `<div class="content">${options.text}</div>`;
```

Comment:
Add `TODO` for potential HTML injection issue

Reply:
Created [WJ-248](https://scuttle.atlassian.net/browse/WJ-248)